scripts/install/SetupDeliveryMonitoring.php

```diff
@@ -37,13 +37,21 @@ class SetupDeliveryMonitoring extends InstallAction
      */
     public function __invoke($params)
     {
-        $pm = $this->getServiceLocator()->get(\common_persistence_Manager::SERVICE_ID);
-        $persistence = $pm->getPersistenceById('default');
+        try {
+            $service = $this->getServiceManager()->get(MonitorCacheService::SERVICE_ID);
+        } catch (ServiceNotFoundException $exception) {
+            $service = new MonitorCacheService(array(
+                MonitorCacheService::OPTION_PERSISTENCE => 'default',
+                MonitorCacheService::OPTION_PRIMARY_COLUMNS => DbSetup::getPrimaryColumns()
+            ));
+            $service->setServiceManager($this->getServiceManager());
+        }
+        $persistence = $service->getPersistence();
+
         DbSetup::generateTable($persistence);
-        $this->registerService(DeliveryMonitoringService::SERVICE_ID, new MonitorCacheService(array(
-            MonitorCacheService::OPTION_PERSISTENCE => 'default',
-            MonitorCacheService::OPTION_PRIMARY_COLUMNS => DbSetup::getPrimaryColumns()
-        )));
+
+        $service->setOption(MonitorCacheService::OPTION_PRIMARY_COLUMNS, DbSetup::getPrimaryColumns());
```

Comment:
this option already set in the constructor

Reply:
fixed